src/proxy_handler.cr

```diff
@@ -25,7 +25,7 @@ class Mitm::ProxyHandler
       host, port = request.resource.split(":", 2)
 
       context.response.upgrade do |io|
-        magic = io.peek.try &.[1]
+        magic = io.peek.try &.[1]?
```

Comment:
If we get zero-length slice by `peek`, this means EOF on the socket.
It's most likely dead, and we should close it.
Although we could wait some time in case something appears.